### PR TITLE
add ability to pass region_name to EmrAddStepsOperator

### PR DESCRIPTION
**Pull request by andriisoldatenko**
possible fix for https://github.com/apache/airflow/issues/13785

### CODE REVIEW

**Review by subkanthi — APPROVED** [review]

File: airflow/providers/amazon/aws/operators/emr_add_steps.py [hunk 1]
```diff
@@ -73,9 +74,10 @@ def __init__(
         self.job_flow_name = job_flow_name
         self.cluster_states = cluster_states
         self.steps = steps
+        self.region_name = region_name
```
Comment (RosterIn) [inline airflow/providers/amazon/aws/operators/emr_add_steps.py]:
```suggestion
        self.region_name = region_name
```

**Review by RosterIn — CHANGES_REQUESTED** [review]

**Review by feluelle — APPROVED** [review]

**Comment by feluelle** [comment]
I re-ran the jobs, because one was failing(, I think unrelated).

**Comment by dstandish** [comment]
> possible fix for #13785

this is not strictly necessary, and not strictly a bug.  region is configurable in the airflow connection (and if using system creds, of course configurable on your system).  

you add kwargs for aws client by putting `config_kwargs` in conn extra: https://github.com/apache/airflow/blob/master/airflow/providers/amazon/aws/hooks/base_aws.py#L371

**Comment by feluelle** [comment]
True. But it is a "possible fix" isn't it?

It still makes sense to configure the region at this stage. We also have done this for other aws operator as far as I remember.

**Comment by dstandish** [comment]
yeah my offering of that solution with the status quo is not meant to indicate oppostion, just saying, you can to this with airflow conn and that's what that user ought to do

having this kindof thing also configurable as params can be convenient and i offer no objection to its addition here :)

small correction on my comment above... as ash pointed out on that issue ... you can put region at top level in conn extra (i think the key is `region_name` and don't need to nest it under `config_kwargs`... a little easier that way

**Pull request closed without merging**